File: Webpage/server/webpage.server/webpage.server/src/main/java/abbruzzese/webpage/webpage/server/webpage/entities/Magazine.java
package abbruzzese.webpage.webpage.server.webpage.entities;


import java.time.LocalDate;
import java.util.UUID;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;

@Entity
public class Magazine {

    public Magazine(){}

    public Magazine(String Nombre,String Descripcion,String Año,String imaString,LocalDate fecha){
        this.nombre=Nombre;
        this.Descripcion=Descripcion;
        this.Año=Año;
        this.imageString=imaString;
        this.fecha=fecha;
    }

    @Id
    private String id=UUID.randomUUID().toString();

    @Column(name="Nombre",nullable=false)
    private String nombre;

    @Column(name="Descripcion",nullable = false)
    private String Descripcion;

    @Column(name="Año",nullable=false)
    private String Año;

    @Column(name="Image",nullable = false)
    private String imageString;

    @Column(name="enlace",nullable=false)
    private String enlace;

    @Column(name="fecha_creacion",nullable=false)
    private LocalDate fecha;

    @PrePersist
    public void prePersist() {
        this.fecha = LocalDate.now(); // Asignar la fecha actual al campo createdAt
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }
    
    public String getNombre() {
        return nombre;
    }

    public void setNombre(String Nombre) {
        nombre = Nombre;
    }

    public String getDescripcion() {
        return Descripcion;
    }

    public void setDescripcion(String descripcion) {
        Descripcion = descripcion;
    }

    public String getAño() {
        return Año;
    }

    public void setAño(String año) {
        Año = año;
    }

    public String getImageString() {
        return imageString;
    }

    public void setImageString(String imageString) {
        this.imageString = imageString;
    }

    public String getEnlace() {
        return enlace;
    }

    public void setEnlace(String enlace) {
        this.enlace = enlace;
    }

    public LocalDate getFecha() {
        return fecha;
    }

    public void setFecha(LocalDate fecha) {
        this.fecha = fecha;
    }

}

File: PHP_API/ENTITIES/Magazine.php
<?php

namespace App\Entity;

use Ramsey\Uuid\Uuid;
use Doctrine\ORM\Mapping as ORM;
use App\Repository\MagazineRepository;

#[ORM\Entity(repositoryClass: MagazineRepository::class)] // Esto usa atributos de PHP 8
#[ORM\Table(name: "magazine")]
class Magazine{

    #[ORM\Id]
    #[ORM\Column(type:"string")]
    #[ORM\GeneratedValue(strategy:"NONE")]
    private $id;

    #[ORM\Column(type:"string",unique: true)]
    private $nombre;

    #[ORM\Column(type:"string")]
    private $descripcion;

    #[ORM\Column(type:"string")]
    private $anio;

    #[ORM\Column(type:"string")]
    private $imagestring;

    #[ORM\Column(type:"string")]
    private $enlace;

    #[ORM\Column(type:"date")]
    private $fecha;
    
    public function __construct($nombre,$descripcion,$año,$imagestring,$enlace) {
        $this->id = Uuid::uuid4();
        $this->nombre = $nombre;
        $this->descripcion = $descripcion;
        $this->anio = $año;
        $this->imagestring = $imagestring;
        $this->enlace = $enlace;
        $this->fecha = \DateTime::createFromFormat('Y-m-d', date('Y-m-d'));;
    }

    //Getters y Setters

    public function getId(){return $this->id;}

    public function getNombre(){return $this->nombre;}
    public function setNombre($nombre){$this->nombre=$nombre;}

    public function getDescripcion() {return $this->descripcion;}
    public function setDescripcion($descripcion) {$this->descripcion = $descripcion;}

    public function getAño() {return $this->anio;}
    public function setAño($año) {$this->anio=$año;}

    public function getimagestring() {return $this->imagestring;}
    public function setimagestring($imagestring) {$this->imagestring = $imagestring;}

    public function getEnlace() {return $this->enlace;}
    public function setEnlace($enlace) {$this->enlace = $enlace;}

    public function getFecha() {return $this->fecha;}
    public function setFecha($fecha) {$this->fecha = $fecha;}

}

?>